### ISSUE TITLE
Trouble with writing mount points of NTFS partitions in hardware-config

### ISSUE THREAD

**Issue by cudakovic** [NixOS/nixpkgs#14624]
## Issue description

I successfully mounted my windows partitions (NTFS) following the steps i described below. However, when I try to generate config, to have my drives automatically mounted when I reboot, i get following error/warning (for each drive): 
warning: don't know how to emit ‘fileSystem’ option for FUSE filesystem ‘/mnt/nameofmydrive’

### Steps to reproduce

1. Create a directory of your preference, as /mnt/windows-partition
2. Mount the desired partition, say /dev/sdn5, in that directory:
    `$ mount /dev/sdn5 `/mnt/windows-partition`
3. Run `nixos-generate-config`. It will update /etc/nixos/hardware-configuration.nix to match the new partition configuration (and configuration.nix stays untouched, unless you use the --force option).

P.S. I have ntfs-3g installed on my system.

## Technical details

* System: NixOS: 16.09pre79453.32b7b00 (Flounder)
* Nix version: nix-env (Nix) 1.11.2
* Nixpkgs version: "16.09pre79453.32b7b00"

**Comment by vcunat** [comment]
Yes, it seems clear the generator has never supported fuse mounts yet.

**Comment by Profpatsch** [comment]
triage: are those mounts supported yet? If not, would you like to create a pull request?

**Comment by TheSirC** [comment]
Sorry to post on such an old post but I suffer from the same exact issue but in 19.03. Is there a workaround that has been found?

**Comment by JohnAZoidberg** [comment]
The workaround is to just add them manually: https://nixos.org/nixos/manual/options.html#opt-fileSystems
I had problems mounting a fuse drive one time but I managed to solve it here: https://github.com/NixOS/nixpkgs/issues/46529

**Comment by TheSirC** [comment]
> I had problems mounting a fuse drive one time but I managed to solve it here: #46529
 Thanks a lot for the reference, I will take a look to implement it on my system.

> This issue is just about `nixos-generate-config` to detect them automatically and add them to your config before you install NixOS.
Rodger that, would there be a way to contribute into that sense ? Is there a following issue (maybe this one then?) on the feature ? What would be the steps for me to implement it and contribute back ?

**Comment by JohnAZoidberg** [comment]
> > This issue is just about nixos-generate-config to detect them automatically and add them to your config before you install NixOS.
>
> Rodger that, would there be a way to contribute into that sense ? Is there a following issue (maybe this one then?) on the feature ? What would be the steps for me to implement it and contribute back ?

Yeah, you can implement that here:
https://github.com/NixOS/nixpkgs/blob/c3a05f90ed4a6c601c94f3bf0f14933c8f965d61/nixos/modules/installer/tools/nixos-generate-config.pl#L349
